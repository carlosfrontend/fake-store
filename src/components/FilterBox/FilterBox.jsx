import PropTypes from 'prop-types';
import styles from './FilterBox.module.css'

export const FilterBox = ({minPrice, handlePriceChange, handleCategoryChange}) => {
    

    return (
        <aside className={styles.filterBox}>
           <div className={styles.priceFilter}>
            <label htmlFor="price">From price:</label>
            <input type="range" id="price" min='0' max='800'  value={minPrice} onChange={handlePriceChange} />
            <span className={styles.priceValue}>${minPrice}</span>
           </div>

          <div className={styles.categoryFilter}>
            <label htmlFor="category">Category:</label>
            <select id="category" onChange={handleCategoryChange}>
              <option value="all">All</option>
              <option value="men's clothing">Men&apos;s Clothing</option>
              <option value="women's clothing">Women&apos;s Clothing</option>
              <option value="jewelery">Jewelery</option>
              <option value="electronics">Electronics</option>
            </select>
          </div>
        </aside>
    )
}

FilterBox.propTypes = {
    minPrice: PropTypes.number,
    handlePriceChange: PropTypes.func,
    handleCategoryChange: PropTypes.func
}