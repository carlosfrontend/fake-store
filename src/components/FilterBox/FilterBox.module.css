.filterBox {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  margin: 2em; 
}

.priceFilter, .categoryFilter {
  display: flex;
  align-items: center;
  gap: 1em;
  font-size: 1.4rem;
  font-weight: bold;
  color: #336688;
  font-family: 'Varela Round', sans-serif;
}

.priceFilter input, .categoryFilter select {
  border-radius: 5px;
  border: 1px solid #336688;
  font-size: 1.2rem;
  color: #336688;
  cursor: pointer;
  font-family: 'Varela Round', sans-serif;
  
}
.priceValue {
  font-weight: bold;
  color: #336688;
}

@media screen and (max-width: 768px) {
    .filterBox {
        flex-direction: column;
        align-items: start;
        gap: 1em;
        margin: 1em;
    }

    .priceFilter, .categoryFilter {
        font-size: 1rem;
    }
    
}