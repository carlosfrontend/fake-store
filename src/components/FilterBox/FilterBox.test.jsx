import { it, describe, expect } from 'vitest';
import { FilterBox } from './FilterBox';
import { render } from '@testing-library/react';


describe('FilterBox Component', () => {
  it('should render the FilterBox component', () => {
    const { container } = render(
      <FilterBox
        minPrice={0}
        handlePriceChange={() => {}}
        handleCategoryChange={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
