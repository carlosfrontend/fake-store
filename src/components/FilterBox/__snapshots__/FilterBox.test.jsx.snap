// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterBox Component > should render the FilterBox component 1`] = `
<div>
  <aside
    class="_filterBox_f1b7d6"
  >
    <div
      class="_priceFilter_f1b7d6"
    >
      <label
        for="price"
      >
        From price:
      </label>
      <input
        data-testid="price-input"
        id="price"
        max="800"
        min="0"
        type="range"
        value="0"
      />
      <span
        class="_priceValue_f1b7d6"
      >
        $
        0
      </span>
    </div>
    <div
      class="_categoryFilter_f1b7d6"
    >
      <label
        for="category"
      >
        Category:
      </label>
      <select
        id="category"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="men's clothing"
        >
          Men's Clothing
        </option>
        <option
          value="women's clothing"
        >
          Women's Clothing
        </option>
        <option
          value="jewelery"
        >
          Jewelery
        </option>
        <option
          value="electronics"
        >
          Electronics
        </option>
      </select>
    </div>
  </aside>
</div>
`;
